packages/app/src/containers/NameInspector.tsx

```diff
@@ -0,0 +1,38 @@
+import { InputField, Spacer } from 'noya-designsystem';
+import { memo, useCallback } from 'react';
+import * as InspectorPrimitives from '../components/inspector/InspectorPrimitives';
+import { useApplicationState } from '../contexts/ApplicationStateContext';
+
+interface Props {
+  names: string[];
+  ids: string[];
+}
+
+export default memo(function NameInspector({ names, ids }: Props) {
```

Comment:
I think I preferred how you had it before, where this component didn't rely on `useApplicationState` at all.

I'd like the inspectors for swatches, layer styles, text styles, and symbols to all be as self-contained as possible, even if it results in some duplicated `<NameInspector />`s, so that they're more reusable.

Also, we should try not to mix `id`s for different kinds of objects. Here an `id` might refer to a swatch or a style, but we have no way to know. I see this is a weakness of TypeScript for not giving us a good way to enforce different id types.

Reply:
the issue I had with this was that I couldn't use the useCallback inside de `SwatchesInspector.tsx` and `LayerStylesInspector.tsx` hook because it gives this error 

> React Hook "useCallback" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?


Because I have a return null, for when the selectedSwatches or StyleLayer it's empty.

Reply:
For cases like that, we can move the `useCallback` above the return, e.g.

```tsx
export default memo(function ComponentsInspectors() {
  /* ... */

  const handleNameChange = useCallback(
    (value: string) =>
      dispatch(
        'setSwatchName',
        selectedSwatches.map((v) => v.do_objectID),
        value,
      ),
    [dispatch, selectedSwatches],
  );

  if (selectedSwatches.length === 0) return null;

  return (
    <Fragment key="layout">
      <NameInspector
        names={selectedSwatches.map((v) => v.name)}
        onNameChange={handleNameChange}
      />
      <ColorPickerInspector />
      <Spacer.Vertical size={10} />
    </Fragment>
  );
});
```

Reply:
ohh! I understand